### ISSUE TITLE
DLS failing to start

### ISSUE THREAD

**Issue by damianday** [d-language-server/dls#7]
OS: Windows 10 x64
DMD2 is in my path
Fresh install of vscode  dls v1.8.1

[Error - 8:31:26 AM] Starting client failed
Error: Unsupported server configuration {
    "command": ""
}
	at _getServerWorkingDir.then.serverWorkingDir (C:\Users\Damian\.vscode\extensions\laurenttreguier.vscode-dls-1.8.1\node_modules\vscode-languageclient\lib\main.js:356:35)
	at <anonymous>

**Comment by LaurentTreguier** [comment]
What's your VSCode install like? Is it stable or insiders, and is it the classic system wide install or the new user level install? (I've never tested the latter)

What happens if you run `dub fetch dls` and `dub run dls:bootstrap --arch=x86_64` in a command prompt?

**Comment by damianday** [comment]
VSCode Stable, it is system wide install.

Visual Code version:
Version: 1.26.1
Commit: 493869ee8e8a846b0855873886fc79d480d342de
Date: 2018-08-16T18:38:57.434Z
Electron: 2.0.5
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

Windows PowerShell
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

PS D:\dlang\Projects\legend> **dub fetch dls**
Please note that you need to use `dub run <pkgname>` or add it to dependencies of your package to actually use/run it. dub does not do actual installation of packages outside of its own ecosystem.
PS D:\dlang\Projects\legend> **dub run --arch=x86_64 dls:bootstrap**
Building package dls:bootstrap in C:\Users\Damian\AppData\Local\dub\packages\dls-0.9.0\dls\bootstrap\
Running pre-generate commands for dls:i18n...
Performing "debug" build using dmd for x86_64.
dls:protocol 0.9.0: building configuration "library"...
dls:i18n 0.9.0: building configuration "library"...
dls:bootstrap 0.9.0: building configuration "application"...
Linking...
D:\dlang\dmd2\windows\bin\lld-link.exe: error: could not open libcmt.lib: no such file or directory
Error: linker exited with status 1
dmd failed with exit code 1.
PS D:\dlang\Projects\legend> **dub run dls**
Building package dls in C:\Users\Damian\AppData\Local\dub\packages\dls-0.9.0\dls\
Fetching msgpack-d 1.0.0-beta.7 (getting selected version)...
Fetching libdparse 0.8.8 (getting selected version)...
Fetching emsi_containers 0.8.0-alpha.9 (getting selected version)...
Fetching dub 1.10.0 (getting selected version)...
Fetching dsymbol 0.3.12 (getting selected version)...
Fetching libddoc 0.3.0-beta.1 (getting selected version)...
Fetching inifiled 1.3.1 (getting selected version)...
Fetching dscanner 0.5.8 (getting selected version)...
Invalid source/import path: C:\Users\Damian\AppData\Local\dub\packages\dcd-0.9.10\dcd\bin
Running pre-generate commands for dcd...
Running pre-generate commands for dls:i18n...
Invalid source/import path: C:\Users\Damian\AppData\Local\dub\packages\dscanner-0.5.8\dscanner\bin
Running pre-generate commands for dscanner...
Running pre-generate commands for dls...
The system cannot find the path specified.
Command failed with exit code 1

**Comment by damianday** [comment]
After running dub run dls:bootstrap it seems to be working fine now for x86, I guess the problem is related to trying to build it with x64

**Comment by LaurentTreguier** [comment]
Yes, apparently the new lld linker has problems with x86_64... It works fine with visual studio build tools.
I'll try to work around this in the extensions.

**Comment by LaurentTreguier** [comment]
The install at first launch should be fixed with vscode-dls 1.9.0

**Comment by Kozzi11** [comment]
Ok for some unknown reason I have not installed dub on my system, but still it should give me better error msg

**Comment by LaurentTreguier** [comment]
Oh. I'll add a check for whether dub is installed just in case then

**Comment by patrickkh7788** [comment]
I try this first time, dcd is download last version. open with a lot errors:
```sh
core.exception.RangeError@/Users/travis/dlang/ldc-1.13.0/bin/../import/std/utf.d(448): Range violation
----------------
??:? pure nothrow @nogc @safe int core.internal.string.numDigits!(10u).numDigits(ulong) [0x1056034f9]
??:? pure nothrow @nogc @safe uint rt.cover.max!(uint).max(uint, uint) [0x10561e96a]
??:? pure nothrow @nogc @safe uint rt.cover.max!(uint).max(uint, uint) [0x10561f65e]
??:? bool core.demangle.mangle!(int function(scope int delegate(immutable(object.ModuleInfo*)))).mangle(const(char)[], char[]).DotSplitter.__xopEquals(ref const(core.demangle.mangle!(int function(scope int delegate(immutable(object.ModuleInfo*)))).mangle(const(char)[], char[]).DotSplitter), ref const(core.demangle.mangle!(int function(scope int delegate(immutable(object.ModuleInfo*)))).mangle(const(char)[], char[]).DotSplitter)) [0x1055fe671]
??:? pure nothrow @nogc core.exception.InvalidMemoryOperationError core.exception.staticError!(core.exception.InvalidMemoryOperationError).staticError() [0x1055fef27]
??:? pure @safe ulong std.utf.toUCSindex!(wchar).toUCSindex(const(wchar)[], ulong) [0x105192d65]
??:? pure @nogc @safe ulong std.algorithm.iteration.reduce!(dls.util.document.Document.byteAtPosition(const(dls.protocol.definitions.Position)).__lambda2).reduce!(ulong, std.range.iota!(const(ulong), const(ulong)).iota(const(ulong), const(ulong)).Result).reduce(ulong, std.range.iota!(const(ulong), const(ulong)).iota(const(ulong), const(ulong)).Result) [0x10518a2ed]
??:? void inifiled.readINIFile!(dscanner.analysis.config.StaticAnalysisConfig).readINIFile(ref dscanner.analysis.config.StaticAnalysisConfig, immutable(char)[]) [0x105082ab0]
??:? std.typecons.Flag!("each").Flag std.algorithm.iteration.each!(dls.tools.analysis_tool.AnalysisTool.scanAllWorkspaces().__lambda1).each!(dls.util.uri.Uri[]).each(dls.util.uri.Uri[]) [0x105081f32]
??:? void dls.protocol.handlers.pushHandler!(void function(std.json.JSONValue)*).pushHandler(const(immutable(char)[]), void function(std.json.JSONValue)*) [0x10509a039]
??:? void dls.protocol.handlers.pushHandler!(void function(std.json.JSONValue)*).pushHandler(const(immutable(char)[]), void function(std.json.JSONValue)*).__lambda3(std.typecons.Nullable!(std.json.JSONValue).Nullable) [0x1050f6f60]
??:? pure nothrow @safe void std.container.dlist.DList!(dls.util.disposable_fiber.DisposableFiber).DList.removeFront() [0x10509f65c]
??:? pure @safe ulong std.conv.to!(ulong).to!(immutable(char)[]).to(immutable(char)[]) [0x10509ed1f]
??:? nothrow void core.thread.thread_scanAll(scope void delegate(void*, void*) nothrow).__lambda2!(core.thread.ScanType, void*, void*).__lambda2(core.thread.ScanType, void*, void*) [0x105607913]
[Info  - 1:14:34 PM] Connection to server got closed. Server will restart.
core.exception.RangeError@/Users/travis/dlang/ldc-1.13.0/bin/../import/std/utf.d(448): Range violation
```
